psiminer-core/src/main/kotlin/psi/method/PHPMethodProvider.kt

```diff
@@ -0,0 +1,49 @@
+package psi.method
+
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import com.jetbrains.php.lang.documentation.phpdoc.psi.impl.PhpDocCommentImpl
+import com.jetbrains.php.lang.psi.elements.impl.MethodImpl
+
+class PHPMethodProvider : MethodProvider() {
+    override fun getNameNode(root: PsiElement): PsiElement =
+        (root as? MethodImpl)?.nameIdentifier
+            ?: throw IllegalArgumentException("Try to extract name not from the method")
+
+    override fun getBodyNode(root: PsiElement): PsiElement? =
+        (root as? MethodImpl)?.lastChild
+            ?: throw IllegalArgumentException("Try to extract name not from the method")
+
+    override fun getDocComment(root: PsiElement): PsiElement? {
+        return PsiTreeUtil.collectElementsOfType(root, PhpDocCommentImpl::class.java).firstOrNull()
```

Comment:
It would be cleaner to use the `PhpDocComment` interface rather than its implementation, as in the future there might be a different implementation of this interface.